### ISSUE TITLE
The network is not convergence when training.

### ISSUE THREAD

**Issue by tangconghui2008** [verlab/accelerated_features#101]
I have downloaded the network from the website and am training it using default parameters, but it does not seem to converge. How can I do?

Loss: 5.3158 acc_c0 0.109 acc_c1 0.493 acc_f: 0.149 loss_c: 3.039 loss_f: 5.681 loss_kp: 1.096 #matches_c: 3551 loss_kp_pos: 7.139 acc_kp_pos: 0.458:  80%|▊| 1279saving iter  1280001 [00:00<?, ?it/s]
Loss: 4.8289 acc_c0 0.646 acc_c1 0.344 acc_f: 0.060 loss_c: 4.895 loss_f: 7.486 loss_kp: 1.354 #matches_c: 433 loss_kp_pos: 9.654 acc_kp_pos: 0.339:  80%|▊| 12845saving iter  128500
Loss: 4.3495 acc_c0 0.367 acc_c1 0.854 acc_f: 0.185 loss_c: 0.963 loss_f: 5.269 loss_kp: 1.151 #matches_c: 1774 loss_kp_pos: 8.887 acc_kp_pos: 0.357:  81%|▊| 1289saving iter  129000
Loss: 4.8264 acc_c0 0.455 acc_c1 0.771 acc_f: 0.155 loss_c: 1.460 loss_f: 5.812 loss_kp: 1.237 #matches_c: 2649 loss_kp_pos: 8.661 acc_kp_pos: 0.376:  81%|

**Comment by tangconghui2008** [comment]
I load the xfeat.pt  to the network when the  begain of training the loss is also not convergence.

**Comment by Odin-byte** [comment]
Have you found any solution or source of error? Do you use work with the preprocessed images (D2-Net) for the megadepth dataset?

**Comment by HoseinAsadolahi** [comment]
I have encountered this issue and then checked the training script and from what I've realized they do not train using an epoch based approach and the accuracies they are reporting is the first and last pair in a batch which for default training is  accuracy for a coco sample and one for megadepth. I tried averaging and making it epoch based and now it is better in case of convergence but the numbers are very low for coarse accuracy(5 percent for validation set and 10 for train set) and I do not know if they are OK or if there is a problem with my training script, if there is, I do not know where it is. I am training only using synthetic data and the data is my own data, so it could be from the data. if you found out anything I would appreciate it if you hit me up with an update.
